drivers/spi/spi_context.h

```diff
@@ -154,32 +154,30 @@ static inline void spi_context_cs_control(struct spi_context *ctx, bool on)
 	}
 }
 
-static inline void spi_context_buffers_setup(struct spi_context *ctx,
-					     const struct spi_buf *tx_bufs,
-					     size_t tx_count,
-					     struct spi_buf *rx_bufs,
-					     size_t rx_count,
-					     u8_t dfs)
+static inline
+void spi_context_buffers_setup(struct spi_context *ctx,
+			       const struct spi_buf_array *tx_bufs,
+			       const struct spi_buf_array *rx_bufs,
+			       u8_t dfs)
 {
-	SYS_LOG_DBG("tx_bufs %p (%zu) - rx_bufs %p (%zu) - %u",
-		    tx_bufs, tx_count, rx_bufs, rx_count, dfs);
+	SYS_LOG_DBG("tx_bufs %p - rx_bufs %p - %u", tx_bufs, rx_bufs, dfs);
 
-	ctx->current_tx = tx_bufs;
-	ctx->tx_count = tx_count;
-	ctx->current_rx = rx_bufs;
-	ctx->rx_count = rx_count;
+	ctx->current_tx = tx_bufs->bufs;
+	ctx->tx_count = tx_bufs->count;
```

Comment:
`tx_bufs` may be `NULL`, so these assignments should be moved inside the `if (tx_bufs)` below.

Same story with `rx_bufs`.